com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -52,6 +53,14 @@ protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exceptio
 
             CloudControllerClient client = execution.getControllerClient();
 
+            boolean deleteServicesFlag = (boolean) execution.getContext()
+                .getVariable(Constants.PARAM_DELETE_SERVICES);
+            
+            if(!deleteServicesFlag) {
+                getStepLogger().warn(Messages.DETECTED_LEFTOVER_SERVICES);
+                return StepPhase.DONE;
+            }
+            
             List<String> servicesToDelete = StepsUtil.getServicesToDelete(execution.getContext());
 

```

Comment:
First check if servicesToDelete is not empty, then:
-if yes - log debug message that there are not services to delete and finish the step
-if not - check the parameter PARAM_DELETE_SERVICES
-- if not - log warn and finish the step
-- if yes -  continue with actual deletion